codegen/src/main/java/software/amazon/awssdk/codegen/poet/paginators/PaginatorsClassSpec.java

```diff
@@ -165,21 +167,24 @@ protected MemberModel memberModelForResponseMember(String input) {
         return shape.getMemberByC2jName(hierarchy[hierarchy.length - 1]);
     }
 
-    protected String hasNextPageMethodBody() {
-        String body;
+    protected CodeBlock hasNextPageMethodBody() {
 
         if (paginatorDefinition.getMoreResults() != null) {
-            body = String.format("return %s.%s.booleanValue()",
+            return CodeBlock.builder()
+                    .add("return $N.$L.booleanValue()",
                                  PREVIOUS_PAGE_METHOD_ARGUMENT,
-                                 fluentGetterMethodForResponseMember(paginatorDefinition.getMoreResults()));
-        } else {
-            // If there is no more_results token, then output_token will be a single value
-            body = String.format("return %s.%s != null",
-                                 PREVIOUS_PAGE_METHOD_ARGUMENT,
-                                 fluentGetterMethodsForOutputToken().get(0));
+                                 fluentGetterMethodForResponseMember(paginatorDefinition.getMoreResults()))
+                    .build();
         }
+        // If there is no more_results token, then output_token will be a single value
+        return CodeBlock.builder()
+                .add("return $1N.$2L != null && !$3T.class.isInstance($1N.$2L) && !$4T.class.isInstance($1N.$2L)",
```

Comment:
So our assumption is "Output token member will not be a list or map" when more_results key is missing in paginators file. Is that correct?

Reply:
No, it doesn't make any assumptions about the output token. We treat auto construct lists and maps as null so it's essentially a null check here just like it was original.